pkg/util/azureclient/mgmt/containerregistry/registries_addons.go

```diff
@@ -0,0 +1,29 @@
+package containerregistry
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	azcontainerregistry "github.com/Azure/azure-sdk-for-go/services/containerregistry/mgmt/2019-06-01-preview/containerregistry"
+)
+
+// RegistriesAddons contains addons for RegistriesClient
+type RegistriesAddons interface {
+	GenerateCredentials(ctx context.Context, resourceGroupName string, registryName string, generateCredentialsParameters azcontainerregistry.GenerateCredentialsParameters) (result azcontainerregistry.GenerateCredentialsResult, err error)
+}
+
+func (r *registriesClient) GenerateCredentials(ctx context.Context, resourceGroupName string, registryName string, generateCredentialsParameters azcontainerregistry.GenerateCredentialsParameters) (result azcontainerregistry.GenerateCredentialsResult, err error) {
+	future, err := r.RegistriesClient.GenerateCredentials(ctx, resourceGroupName, registryName, generateCredentialsParameters)
+	if err != nil {
+		return
+	}
+
+	err = future.WaitForCompletionRef(ctx, r.Client)
+	if err != nil {
+		return
+	}
+	result, err = r.RegistriesClient.GenerateCredentialsResponder(future.Response())
```

Comment:
@jim-minter this is all that was missing. I think the api is correct, it's just unusual to have an async method that returns data. so you have to wait on the future then call the responder.